crates/engine/tree/src/tree/payload_processor/mod.rs

```diff
@@ -492,6 +531,60 @@ impl ExecutionCache {
     }
 }
 
+/// Container for receiving end of executable transactions stream.
+#[derive(Debug)]
+pub struct ExecutableTxReceiver<T> {
+    rx: mpsc::Receiver<T>,
+    len: usize,
+}
```

Comment:
I think if we want to do smth like

payload -> (env, iter, remainder)

then this iterator must yield results, because the iter must handle the Bytes -> Tx transformation